plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlScannerTest.java

```diff
@@ -17,19 +17,64 @@
 package com.google.cloud.tools.eclipse.appengine.validation;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
 
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+
+import org.eclipse.core.resources.IContainer;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.JavaModelException;
 import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
 import org.junit.Test;
 import org.xml.sax.SAXException;
 import org.xml.sax.ext.Locator2Impl;
 import org.xml.sax.helpers.AttributesImpl;
 
+import com.google.cloud.tools.eclipse.test.util.project.TestProjectCreator;
+
 public class WebXmlScannerTest {
 
-  private WebXmlScanner scanner = new WebXmlScanner();
+  private static IFile resource;
+  private static IFile servletClass;
+  private static IJavaProject javaProject;
+  private WebXmlScanner scanner;
+  
+  @ClassRule public static TestProjectCreator projectCreator = new TestProjectCreator().withClasspathContainerPath("src/main/java");
```

Comment:
Let's remove `withClasspathContainerPath()`. You're in the right path in that you will do something with project classpath entries to designate Java source folders, but this line is not going to work as you hoped.